### PR TITLE
smoketests: updates for prioritization work

### PR DESCRIPTION
**Pull request by mastercactapus**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**
This PR updates the smoketests to allow for more time in-between steps to allow for future rate-limiting without causing test failures.

- 1-minute increments have largely been replaced with 30-minute increments
- WaitAndAssert now waits for extra engine cycles instead of a fixed-time-delay
- extra data from a few tests (e.g. creating alerts when notifications were not being tested) cleaned up
- WaitAndAssert forces engine cycles and fast-forwards time (e.g. simulate a 5-minute wait for no-more messages instead of 15 seconds)

### CODE REVIEW

File: smoketest/addrules_test.go [hunk 1]
```diff
@@ -71,18 +71,18 @@ func TestAddRules(t *testing.T) {
 	d.ExpectSMS("testing")
 	tw.WaitAndAssert()
 
-	h.FastForward(time.Minute)
+	h.FastForward(30 * time.Minute)
```
[the remark this answers is not in the document]
Comment (mastercactapus) [reply smoketest/addrules_test.go]: We do use the same value in most places, but they are actually specific to each test event rather than being a special/generic value that we'd want to put in a constant.

All the generic fast-forward stuff is done in the WaitAndAssert logic, but each one of these is determined by the init SQL for that test, which could change with each test and frequently does when debugging. For example, in the SQL code we may specify 30-minute rules, or 15 and 45, so there is a need to FF 30 minutes. The 30-minute value only has meaning because of what was configured for that test and is important when reading the test to see how far time is advancing.

Basically it's convention driving the same 30 minutes for each, and not something we could change globally without rewriting/updating each individual test. Plus, knowing how much FF each step of the way is super helpful when debugging.

File: smoketest/inprogress_test.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ func TestInProgress(t *testing.T) {
     insert into notification_policy_cycles (alert_id, user_id, last_tick)
     values
         (1, {{uuid "u1"}}, null),
-        (1, {{uuid "u2"}}, now() + '5 minutes'::interval),
+        (1, {{uuid "u2"}}, now() + '35 minutes'::interval),
```
[the remark this answers is not in the document]
Comment (mastercactapus) [reply smoketest/inprogress_test.go]: The old version of this test had the notification rules set to trigger after 1 minute, and cycle starrt was set to have started 5-minutes in the future (thus 4 minutes-past the rules). Basically it is set up so the cycle start is AFTER all the rules would have finished. If we wanted to keep it the same "distance" we could do 34, but all that matters is that it is at least a minute or two beyond the last notification rule.
